src/metadata/persistence/templates.clj

```diff
@@ -264,15 +329,9 @@
           (where {:id template-id}))
   nil)
 
-;; TODO: depending on how synonyms are going to be implemented, it may be necessary to check for existing synonyms
-;; before deleting the attributes as well.
-(defn- used-attribute-subselect
-  []
-  (subselect [:template_attrs :ta] (where {:ta.attribute_id :attributes.id})))
```

Comment:
Since the code to update templates already checked for attribute synonyms, I decided to code it to check for synonyms as well. There could be some consequences of this that I haven't thought of, though.